sdk/communication/AzureCommunicationChat/Source/PushNotification/AppGroupPushNotificationKeyHandler.swift

```diff
@@ -0,0 +1,94 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+
+public class AppGroupPushNotificationKeyHandler: PushNotificationKeyHandler {
+    var sharedDefault: UserDefaults
+    var keyTag: String
+
+    public init?(appGroupId: String, keyTag: String) {
+        guard let shareDefault = UserDefaults(suiteName: appGroupId) else {
+            return nil
+        }
+        self.sharedDefault = shareDefault
+        self.keyTag = keyTag
+    }
```

Comment:
To me this seems like it should be a throwing initializer. If this fails, it gives no indication why, when the reason is pretty clear--the UserDefaults don't exist for the app group ID. It would be better to throw and give the user an actionable error. If they do want this behavior they can just do `let handler = try? AppGroupPushNotificationKeyHandler(...)`

Reply:
agree. This is relevant to my answer to the previous comment. I will add more error handling logics in key management part.

Reply:
Change pushed. API View updated.